### PR TITLE
Add Confirmation Email for Crypto Deposit

### PR DESCRIPTION
**Pull request by SudharakaP**
⚠️ Should be merged only after https://github.com/opencollective/opencollective-api/pull/6411

Related to https://github.com/opencollective/opencollective/issues/4527

image

### CODE REVIEW

**Review by znarf — CHANGES_REQUESTED** [review]
Don't reuse the function and the template. Use a separated function and a separated template.

**Comment by SudharakaP** [comment]
> Don't reuse the function and the template. Use a separated function and a separated template.

Thanks. I've refactored it to a separate function and a separate template. 👍🏽

File: server/lib/payments.js [hunk 1]
```diff
@@ -590,6 +592,28 @@ const sendOrderConfirmedEmail = async (order, transaction) => {
...
+  const { collective, fromCollective } = order;
+  const user = order.createdByUser;
+  const host = await collective.getHostCollective();
+
+  const data = {
+    order: order.info,
+    collective: collective.info,
+    host: host.info,
+    fromCollective: fromCollective.activity,
+    isCrypto: true,
```
Comment (znarf) [inline server/lib/payments.js:607]: I don't think that's needed.

File: templates/emails/order.crypto.processing.hbs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+{{/if}}
+
+<p>Thank you for your crypto contribution to {{collective.name}} in the amount of {{pledgeAmount}} {{pledgeCurrency}}, made on
+  {{moment order.createdAt}}.</p>
+
+<p>Your contribution is currently pending. Once you transmit the funds,
+   and they have been confirmed as received by <a href="https://opencollective.com/{{host.slug}}">{{host.name}}</a>,
+   the amount will be credited to {{collective.name}}. </p>
+
+<p>Note that we use <a href="https://www.thegivingblock.com/">Giving Block</a>
```
Comment (znarf) [inline templates/emails/order.crypto.processing.hbs:30]: It's not necessary to mention The Giving Block.

File: server/lib/payments.js (hunk 1 above)
Comment (SudharakaP) [reply server/lib/payments.js]: Yep, my mistake, I forgot to remove this after refactoring. Done. 👍🏽

File: templates/emails/order.crypto.processing.hbs (hunk 2 above)
Comment (SudharakaP) [reply templates/emails/order.crypto.processing.hbs]: Thanks, I've removed this paragraph. 👍🏽

**Review by znarf — CHANGES_REQUESTED** [review]
We talked about having the deposit address or the QR code in the email.

QR code might be difficult but can we add the address somewhere?

**Comment by SudharakaP** [comment]
> We talked about having the deposit address or the QR code in the email.
> 
> QR code might be difficult but can we add the address somewhere?

@znarf : Ops, something I've missed. Yes I've added the deposit address in the above email. Let me know if it looks good. The screenshot above is updated. 🤔

**Comment by SudharakaP** [comment]
Merged as per https://opencollective.slack.com/archives/C0RMV6F8C/p1629475803098700

**Pull request merged**